src/SearchComponents/SearchField.svelte

```diff
@@ -1,21 +1,45 @@
 <script>
     /* Assumes text property */
     import {getLabelForProp, getListByPropName, getObjectTypeForProp} from "../dsp-services";
-    import {getFilterByNameAndVal, getPropString} from './SearchUtility';
-    export let value;
-    export let prop;
-    export let shortName;
+    import {getFilterByNameAndVal, getPropString, getPropStringHelper} from './SearchUtility';
+
+    export let parent;
+    export let prop, predefProp, predefVal;
     let dateDepth;
+    let children = [];
+    let value = "";
     let name = prop["propName"]; // helper to avoid refreshing on await command in html.
     import {language} from '../store.js'
+    import SearchField from "./SearchField.svelte";
 
     /*
         This function return the gravsearch string that references this value. It takes into account whether its a property of the main resource or if the main resource is linked.
         @return: the gravsearch string
     */
     export function getString() {
-
-        return getPropString(prop);
+        let toReturn = getPropStringHelper(prop, parent);
+        if (children.length > 0){
+            for (const child of children){
+                toReturn += child.getString();
+            }
+        }
```

Comment:
Is this if clause necessary? As far as I understood the for loop only starts if the children string has as least on character.

Reply:
agreed